app/javascript/controllers/tag_select_controller.js

```diff
@@ -50,4 +55,14 @@ export default class extends Controller {
       this.tagContainer.classList.add("is-changed")
     }
   }
+
+  dedupeValues(values) {
+    const sanitizedVals = values.map((val) => {
+      return val.trim().toLowerCase()
+    })
+    const dedupedVals = [...new Set(sanitizedVals)]
+    return dedupedVals.filter((val) => {
```

Comment:
If there were null/undefined values in here, wouldn't line 61 (`null.trim().toLowerCase()`) already have errored?  Maybe you should do this first?

A simple falsey check also works, since empty strings are false in javascript `values.filter(val => val)`